lib/prefixed_ids/prefix_id.rb

```diff
@@ -13,25 +13,34 @@ def initialize(model, prefix, salt: PrefixedIds.salt, minimum_length: PrefixedId
     def encode(id)
       return if id.nil?
 
-      @prefix + @delimiter + @hashids.encode(TOKEN, id)
+      @prefix + @delimiter + @hashids.encode([TOKEN] + Array.wrap(id))
     end
 
     # decode returns an array
     def decode(id, fallback: false)
       fallback_value = fallback ? id : nil
       _, id_without_prefix = PrefixedIds.split_id(id, @delimiter)
       decoded_hashid = @hashids.decode(id_without_prefix)
+
       if fallback && !valid?(decoded_hashid)
         fallback_value
+      elsif !valid?(decoded_hashid)
+        nil
       else
-        decoded_hashid.last
+        _ = decoded_hashid.shift
+
+        if decoded_hashid.size == 1
+          decoded_hashid.first
+        else
+          decoded_hashid
+        end
```

Comment:
I think we can use array destructuring here to clean this up as well:
```suggestion
        _, id, *composite = decoded_hashid
        composite.empty? ? id : composite
```